### PR TITLE
Update the shipping label status API to fetch multiple ones

### PR DESCRIPTION
**Pull request by hichamboushaba**
## Description
The purchase label endpoint is asynchronous (the sync behavior is [discouraged](https://href.li/?1227-gh-Automattic/woocommerce-connect-server), and the web client uses the status endpoint for each created label individually after the purchase.
With the mobile apps consuming the API, there are some new constraints regarding the network availability, and the round-trip delays, so this PR is updating the `label status` endpoint to accept multiple label_ids at the same time, and updates the web client to use it instead, and following the list of changes:
1. Update the endpoint to accept multiple ids: 6bd250a
2. Update the labels polling after a purchase to use the new endpoint: 3c3ae4f.
3. Update the logic of fetching labels to use the new endpoint: 369e0b8

Relevant discussion: p91TBi-4K3-p2

### Related issue(s)

https://github.com/woocommerce/woocommerce-android/issues/3698

### Steps to reproduce & screenshots/GIFs

- Manually test the label status endpoint, and confirm that the response has a `labels` array, containing the data of the passed ids.
- Make shipping label purchases, and confirm that there is no regression.
- Confirm that there is no regression when loading an order that has shipping labels. 

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added

### CODE REVIEW

**Comment by hichamboushaba** [comment]
A question to the reviewer, do we need to update the code in `wp-calypso` repo: https://github.com/Automattic/wp-calypso/tree/trunk/client/extensions/woocommerce/woocommerce-services to use the new endpoint as well?

**Review by frosso — APPROVED** [review]
I tested this with single and multiple packages. Works great!

File: classes/class-wc-rest-connect-shipping-label-status-controller.php [hunk 1]
```diff
@@ -9,27 +9,30 @@
...
+			$label    = $this->settings_store->update_label_order_meta_data( $request['order_id'], $response->label );
+			$labels[] = $label;
 		}
 
-		$label = $this->settings_store->update_label_order_meta_data( $request['order_id'], $response->label );
-
 		return array(
 			'success' => true,
-			'label'   => $label,
+			'labels'  => $labels,
```
[the remark this answers is not in the document]
Comment (hichamboushaba) [reply classes/class-wc-rest-connect-shipping-label-status-controller.php:35]: Updated in 1fda5c8.
> As an alternative for still supporting older clients, we could still return label with the first element from the labels array. Pretty minor tho, I think.

Unless there are other clients which can be impacted, I think keeping only `labels` is cleaner, and will avoid an unnecessary increase of the payload size 🙂, WDYT?

File: client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js [hunk 2]
```diff
@@ -1108,49 +1108,41 @@ export const openRefundDialog = ( orderId, siteId, labelId ) => {
...
+		} )
+		.catch( error => {
+			dispatch( {
+				type: WOOCOMMERCE_SERVICES_SHIPPING_LABEL_STATUS_RESPONSE,
+				orderId,
+				siteId,
+				labelsData: null,
+				error,
+			} );
+			throw error;
```
[the remark this answers is not in the document]
Comment (hichamboushaba) [reply client/extensions/woocommerce/woocommerce-services/state/shipping-label/actions.js]: Done in 864d0f0, thanks for the input 🙏

**Comment by hichamboushaba** [comment]
Thanks @frosso for the review, I tried addressing your comments, I'll hold on merging the PR until you take a second look 🙂 .

File: classes/class-wc-rest-connect-shipping-label-status-controller.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (frosso) [reply classes/class-wc-rest-connect-shipping-label-status-controller.php:35]: That sounds great!

**Review by frosso — APPROVED** [review]
Thanks for the updates @hichamboushaba!
Would it be ok for you to wait until https://github.com/Automattic/woocommerce-services/pull/2399 is merged, so that the changelog can be updated in this PR?

If you'd like to, I can keep an eye on it and just update this PR once the other one is merged. so that it's off your plate!

**Comment by hichamboushaba** [comment]
> Would it be ok for you to wait until #2399 is merged, so that the changelog can be updated in this PR?

Sure, I can wait 🙂, I'll update the changelog after it's merged.

**Comment by hichamboushaba** [comment]
@frosso just to not forget about this, can you please answer my question here: https://github.com/Automattic/woocommerce-services/pull/2390#issuecomment-805787737 ?

**Comment by frosso** [comment]
@hichamboushaba Thanks for pointing it out, I forgot to reply to that! We should be ok, since the Calypso components are getting deprecated!

**Comment by Ferdev** [comment]
Hey all! Just wanted to let you know #2399 has been merged into develop.

**Pull request merged**